clients/client-dart/lib/src/json_decoder.dart

```diff
@@ -10,22 +10,12 @@ class JsonDecoder extends MessageDecoder<String> {
     
     var event_as_list = jsonDecode('{"received": $event }')['received'];
 
-    var msg = ChannelMessage(Utils.checkString(event_as_list[0]), 
+    return ChannelMessage(Utils.checkString(event_as_list[0]), 
       Utils.checkString(event_as_list[1]),
       Utils.checkString(event_as_list[2]),
-      null);
-    
-    var data = Utils.checkString(event_as_list[3]);
-
-    if (data != null) {
-      try {
-        msg.payload = jsonDecode(data);
-      } catch (e) {
-        msg.payload = data;
-      }
-    }
-
-    return msg;
+      Utils.checkString(event_as_list[3])
```

Comment:
The channel sender can send a serializable object, if this is the case this line failes because dart interprets this object how hashmap and this is not a subtype of String

Reply:
Leaved the `event_as_list[3]` as a such and no string checking is performed. See PR.

Reply:
This is ok, listeners have to process dynamic payload. 

Thank you @gabheadz